inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.h

```diff
@@ -29,26 +30,36 @@ class MKLDNNPadNode : public MKLDNNNode {
         SYMMETRIC = 3
     };
 
-    void padConstantOrEdge(const float *srcData, float* dstData,
-                           InferenceEngine::SizeVector srcDims, InferenceEngine::SizeVector dstDims,
-                           const bool isEdge = false);
-    void padReflectOrSymmetric(const float *srcData, float* dstData,
-                               InferenceEngine::SizeVector srcDims, InferenceEngine::SizeVector dstDims,
-                               const bool isSymmetric = false);
+    template<typename T> void padConstant();
+    void padConstantZero();
+    void padEdge();
+    void padReflectOrSymmetric(const bool isSymmetric = false);
 
-    size_t getWorkAmountDst() const;
+    void prepareParams();
+
+    inline size_t parallel_init(size_t start, std::vector<size_t> &indexes);
```

Comment:
It is better to rewrite all such auxiliary functions as `static inline` and remove from the header.